keps/262-ConfigurableFailurePolicy/README.md

```diff
@@ -0,0 +1,392 @@
+nodes, and one which runs auxilliary software on a CPU pool. If a child job in the auxilliary software ReplicatedJob 
+crashes, I just want to restart that particular job. If a child job of the GPU/TPU training ReplicatedJob crashes, I want
+to restart the entire JobSet and resume training from the latest checkpoint.
+
+**Example Failure Policy configuration for this use case**:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: configurable-failure-policy-2
+  annotations:
+    alpha.jobset.sigs.k8s.io/exclusive-topology: {{topologyDomain}} # 1:1 job replica to topology domain assignment
+spec:
+  # Failure Policy configured to restart any failed child job in the cpu-worker ReplicatedJob, but fall back to the
+  # default behavior of restarting the entire JobSet if any child job in the tpu-worker ReplicatedJob fails.
+  failurePolicy:
+    rules:
+    - action: RestartJob
+      targetReplicatedJobs: cpu-worker
+    maxRestarts: 10
+  replicatedJobs:
+  - name: gpu-worker 
+    replicas: 4 # set to number of node pools
+    template:
+      spec:
+        parallelism: 2 # number of nodes per pool
+        completions: 2 # number of nodes per pool
+        backoffLimit: 0
+        template:
+          spec:
+            containers:
+            - name: main
+              image: pytorch:latest
+              command: ["..."]
+            resources:
+              limits:
+                nvidia.com/gpu: 1
+  - name: gpu-worker 
+    replicas: 1 # set to number of node pools
+    template:
+      spec:
+        parallelism: 1 # number of nodes per pool
+        completions: 1 # number of nodes per pool
+        backoffLimit: 0
+        template:
+          spec:
+            restartPolicy: Never
+            containers:
+            - name: main
+              image: python:3.10
+              command: ["..."]
+              
+```
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Proposed Failure Policy API
+
+```go
+
+// FailurePolicyAction defines the action the JobSet controller will take for
+// a given FailurePolicyRule.
+type FailurePolicyAction string
+
+const (
+  // Fail JobSet immediately, regardless of maxRestarts.
+  FailJobSet FailurePolicyAction = "FailJobSet"
+
+  // Don't count the failure against maxRestarts.
+  Ignore FailurePolicyAction = "Ignore"
```

Comment:
can we add a story for this case

Reply:
Added

Reply:
Maybe it should be `IgnoreAndRestartJobSet`?

It's not clear what happens with the jobs that didn't directly fail.